### DISCUSSION TITLE
Will this work with node.js < v18?

### DISCUSSION THREAD

**Discussion by hanebuechenes** [team-exor/eiquidus#39] [General]
Because i have an old coin only working with ubuntu xenial so i can install node.js v20....

**Comment by joeuhren** [comment]
If you follow the installation instructions in the README, you can use `nvm` to install any version of node.js on any system. It is typically recommended to install the latest LTS release. If you want to install an older version for some reason, keep in mind that the lowest supported version is v16.20.1.

**Comment by hanebuechenes** [comment]
Thanks for your hint. So i installed node with nvm and it worked. (Now using node v18.19.0)
Got some troubles with the right mongodb Version..
But at the and, i got stuck with missing libs in the step of installing the modules into the project...

node: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by node)
node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by node)
node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)

Is it possible, to use the explorer on another server than my coind ?

**Comment by joeuhren** [comment]
Thanks for the feedback. It seems you can install any node version on any system you want but some older systems still require special treatment. A quick search of your error brought up this: https://stackoverflow.com/a/74760618/3038650

> This error happens when you install Node on an operating system that has a version of GLIBC that is lower than Node can support. For example, Node v18 supports GLIBC v2.7 or later. When you attempt to install Node v18.x on any Linux OS that has GLIBC v2.6 or lower, you will get such errors.
> 
> To see the version of GLIBC that your operating system has, execute the ldd command on the terminal:
> 
> $ ldd --version
> 
> I can suggest three options to resolve this issue:
> 
>     Either install Node from source code. This will allow you to recompile
>     Or install an older version of Node that your OS can support
>     Or upgrade your Linux OS to a newer version.

Out of those 3 options, the one that might still work for you is to install an older version of Node. As I mentioned earlier, the explorer should still work with v16.20.1. You can try using nvm to install that version and see if it works.

> Is it possible, to use the explorer on another server than my coind ?

Yes, this is possible although it's a bit more complicated since it requires you to open up your coind wallet to another ip address properly which is outside the scope of setup and support of the explorer.

**Comment by hanebuechenes** [comment]
Got everything up and working... 
But then...

npm run sync-blocks
Script launched with pid: 2602
Syncing blocks.. Please wait..
Error: Unable to connect to explorer API

this works:
curl --user kushrpc2 --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "getinfo", "params": [] }' -H 'content-type: text/plain;' http://XX.XX.XX.XX:31543/

any ideas?

**Reply by joeuhren** [reply]
At the beginning of the block sync it calls the getblockcount api and if the return value is not a positive number 0 or greater then you get the "Unable to connect to explorer API" error. This could happen for various reasons. Usually its because your wallet settings aren't set up correctly in settings.json so definitely double-check that part as maybe you mistyped the username or password for example. Another common mistake is using the P2P port instead of the RPC port. Since it seems you were able to successfully connect using curl, it could be that you didn't let your wallet fully load as most wallets will return a block count of -1 before it has finished loading. One other thing I can think of is that maybe your wallets cmd for getting the latest block isn't `getblockcount`? If so, you can change the cmd in the settings.json in the api_cmds section.

**Comment by hanebuechenes** [comment]
OMG, it is working! Thanks bro, this is so nice to see! This is a real good piece of work, i like it a lot!

Thanks for fast support and help!

I just had to modify the api_cmds in settings.json and all was working nicely!

**Comment by joeuhren** [comment]
Glad to hear it! I've been thinking of adding some kind of wallet_test feature that can test all the different wallet cmds that the explorer uses to quickly troubleshoot which ones don't work and help determine when it may be necessary to change certain api_cmds in the settings. Each coin is slightly different and as time goes on there are more variations so its difficult to keep up =)
